lib/stripe/api_resource.rb

```diff
@@ -23,7 +28,7 @@ def self.resource_url
       end
       # Namespaces are separated in object names with periods (.) and in URLs
       # with forward slashes (/), so replace the former with the latter.
-      "/v1/#{self::OBJECT_NAME.downcase.tr('.', '/')}s"
+      "/v1/#{object_name.downcase.tr('.', '/')}s"
```

Comment:
I'm slightly worried about this use case:

Somebody defines their own StripeResource (I've fairly sure that we have occasionally instructed users to do this when they have access to / want to use APIs that are private)
```ruby
class SecretLlama < APIResource
  OBJECT_NAME = "secret_llama"
  include Stripe::APIOperations::Create
end
```
I suspect this change would cause `SecretLlama.create()` to hit `POST /v1/s` instead of `/v1/secret_llamas`.

Reply:
Your suspicion is definitely correct (I ran into this in my testing), though I suspect that this will make it difficult to turn on sorbet static typechecking. Let me consult the sorbet team.

Reply:
Result: we're using `const_get` as a fallback until we can deprecate the constants. I've added a test case to cover this as well!